flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -751,17 +751,28 @@ private void initializeCatalogs() {
     }
 
     private StreamExecutionEnvironment createStreamExecutionEnvironment() {
-        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        // We need not different StreamExecutionEnvironments to build and submit flink job,
+        // instead we just use StreamExecutionEnvironment#executeAsync(StreamGraph) method
+        // to execute existing StreamGraph.
+        // This requires StreamExecutionEnvironment to have a full flink configuration.
+        final StreamExecutionEnvironment env =
+                new StreamExecutionEnvironment(new Configuration(flinkConfig), classLoader);
```

Comment:
I suggest to use `StreamExecutionEnvironment.getExecutionEnvironment(flinkConfig)` instead.

Reply:
It seems planner will configure the `ExecutionEnvironment` when translate the sql to plan.

Reply:
After discussion with Jark offline, currently it's not able to use the `StreamExecutionEnvironment.getExecutionEnvironment(flinkConfig)` to build the `StreamExecutionEnvironment`. Because it doesn't allow us to use the user-defined classloader to build the expected `StreamExecutionEnvironment`.

In my local testing, the used way doesn't have problems(standalone).